src/main/java/net/snowflake/client/core/SFSession.java

```diff
@@ -438,6 +440,18 @@ public void addSFSessionProperty(String propertyName, Object propertyValue) thro
           }
           break;
 
+        case MAX_JSON_STRING_LENGTH:
+          if (propertyValue != null) {
+            maxJsonStringLength = (Integer) propertyValue;
+
+            // Update the max string length used in ObjectMapperFactory
+            ObjectMapperFactory.setMaxJsonStringLength(maxJsonStringLength);
```

Comment:
`maxJsonStringLength` is a variable in the `SFSession` which seems to be specific for a one session but we are changing the static (global) factory and static object mapper causing all the next sessions to use the last configuration (or fallback to the default as changed `DefaultSfConnectionHandler`) - it may be misleading

Also there are many other places where `ObjectMapperFactory` is used and we are not reconfiguring the mappers created from the factory

I think we should keep the final configuration of variable `maxJsonStringLength` only in the factory since it is a global state but we are using the factory in a static way

I am thinking how to make it configurable in the easiest way and how can it be used by the users:

1) we can programatically set the `maxJsonStringLength` on session and ensure that we are reusing the same object mapper in the whole code of the session - it may lead us to a lot of refactoring now
2) sometimes connection string is a user way for creating connection to SF and some frameworks take care of setting up all the driver classes so configuration variable as you created looks good, but we need to ensure that the point (1) is fullfilled
3) There is another way how we can configure the object mapper factory in one place - we can use java property e.g. we can pass `-Dnet.snowflake.client.jdbc.objectMapper.maxJsonStringLength=45000000` and read this variable using `SnowflakeUtil.systemGetProperty` in `ObjectMapperFactory` during `objectMapper` creation
 
Personally, I think the last option will touch all the places instantly and can be easily used by users when no other programatic options are available to change driver's configuration

Please share your thoughts @sfc-gh-ext-simba-jl @sfc-gh-igarish @sfc-gh-pfus